structlog_sentry/__init__.py

```diff
@@ -0,0 +1,39 @@
+import logging
+import os
+import sys
+
+from sentry_sdk import capture_exception, capture_message
+
+
+class SentryProcessor:
+    """Sentry processor for structlog. Uses Sentry SDK to capture events in Sentry."""
+
+    def __init__(self, sentry_log_level=logging.ERROR):
+        self.sentry_log_level = sentry_log_level
+
+    def _log(self, event_dict, level="error"):
+        exc_info = event_dict.pop("exc_info", sys.exc_info())
+        has_exc_info = exc_info and exc_info != (None, None, None)
+
+        if has_exc_info and logging.getLevelName(level.upper()) >= logging.ERROR:
+            return capture_exception(error=exc_info)
+        return capture_message(event_dict.get("event", ""), level=level)
+
+    def __call__(self, logger, method, event_dict):
+        event_dict["sentry"] = "skipped"
+
+        sentry_skip = event_dict.pop("sentry_skip", False)
+        sentry_dsn = os.environ.get("SENTRY_DSN")
```

Comment:
I would remove constraint that there has to be `SENTRY_DSN` in environment. Users of this `SentryProcessor` may have different ways how to pass DSN into app.

Technically you don't want to put `SentryProcessor` into Structlog config if there is not DNS set. But it may not be a nice code as the order of processors matters and you would need to conditionally put it in the middle of the list.

Let's add `active=True` parameter into `__init__`. And in Structlog config it may be used for example as `SentryProcessor(active=bool(os.environ.get("SENTRY_DSN")))`. Or may be deactivated because of some debug settings. It's more flexible for different use cases.

Reply:
OK, that's a good point.